src/main/java/com/synopsys/integration/blackduck/service/request/NotificationRequestFactory.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * blackduck-common
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.blackduck.service.request;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.synopsys.integration.blackduck.api.core.response.BlackDuckPathMultipleResponses;
+import com.synopsys.integration.blackduck.api.generated.discovery.ApiDiscovery;
+import com.synopsys.integration.blackduck.api.manual.temporary.enumeration.NotificationType;
+import com.synopsys.integration.blackduck.api.manual.view.NotificationView;
+import com.synopsys.integration.blackduck.http.BlackDuckRequestBuilder;
+import com.synopsys.integration.blackduck.http.BlackDuckRequestFactory;
+import com.synopsys.integration.blackduck.http.BlackDuckRequestFilter;
+import com.synopsys.integration.rest.RestConstants;
+
+public class NotificationRequestFactory {
+    private final List<String> ALL_NOTIFICATION_TYPES = Stream.of(NotificationType.values()).map(Enum::name).collect(Collectors.toList());
```

Comment:
Does it make sense to have the factory used in NotificationService now? I suspect there's some mildly similar code.